src/zarr/core/metadata/v3.py

```diff
@@ -304,17 +314,17 @@ def parse_fill_value(
     ----------
     fill_value: Any
         A potential fill value.
-    dtype: BOOL_DTYPE | INTEGER_DTYPE | FLOAT_DTYPE | COMPLEX_DTYPE
+    dtype: BOOL_DTYPE | INTEGER_DTYPE | FLOAT_DTYPE | COMPLEX_DTYPE | DATETIME_DTYPE
```

Comment:
I'm not sure we can/should support the datetime type in the v3 metadata yet. This is intended to be an extension type (https://github.com/zarr-developers/zarr-specs/issues/110, https://github.com/zarr-developers/zarr-specs/issues/273) but has not been finalized (or even proposed in sufficient detail). 

@d-v-b / @normanrz / others -- what do we want to do here? This is an example of a regression from v2 that we could handle in a few ways. I think the best thing is to try to replicate support for datetime types in v2 metadata but raise an error if a user creates this type in v3 for now.

Reply:
I spoke with @d-v-b about this today. The action item for the `datetime` dtype is to error/warn in `parse_dtype` is not supported by the spec.

Reply:
+1 to warning / erroring on dtypes that are not defined in the spec. 

We should 100% aim to support datetime types (because people use them), and so I think we should look at what language we can change in the spec to make this viable. I think the simplest approach would be twofold:
- introduce language in the spec distinguishing between core dtypes and community dtypes. core dtypes are part of the core spec, community dtypes come from anyone. the datetime dtype would be a community dtype.
- write up a simple spec for the datetime dtype sufficient for non-python implementation to implement it. submit that document to the collection of community dtypes. 

with that in place, i think we can safely support the dtype with minimal risk of friction across zarr implementations

Reply:
> I think datetime is a particular complex dtype, because there are many different ways to encode datetime in binary form (e.g. precision, timezones, reference date). We should aim on defining an dtype encoding that is either very explicit in its name or has broad community support.

Definitely. I'm hoping that whatever was the norm in zarr v2 can be documented quickly enough to make the transition smooth, but longer term this particular dtype might not be a winner for cross-platform support.

Reply:
see https://github.com/zarr-developers/zarr-python/pull/2200